plugins/plugin-chart-echarts/src/Graph/controlPanel.tsx

```diff
@@ -0,0 +1,241 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { t } from '@superset-ui/core';
+import {
+  formatSelectOptions,
+  sections,
+  sharedControls,
+  ColumnOption,
+  ColumnMeta,
+} from '@superset-ui/chart-controls';
+import { DEFAULT_FORM_DATA } from './types';
+import {
+  legendMarginControl,
+  legendOrientationControl,
+  legendTypeControl,
+  showLegendControl,
+} from '../controls';
+
+const noopControl = { name: 'noop', config: { type: '', renderTrigger: true } };
+const controlConfig = {
+  ...sharedControls.entity,
+  type: 'SelectControl',
+  multi: false,
+  freeForm: true,
+  default: null,
+  includeTime: false,
+  optionRenderer: (c: ColumnMeta) => <ColumnOption column={c} showType />,
+  valueRenderer: (c: ColumnMeta) => <ColumnOption column={c} />,
+  valueKey: 'column_name',
+  allowAll: true,
+  filterOption: ({ data: opt }: any, text: string) =>
+    (opt.column_name && opt.column_name.toLowerCase().indexOf(text.toLowerCase()) >= 0) ||
+    (opt.verbose_name && opt.verbose_name.toLowerCase().indexOf(text.toLowerCase()) >= 0),
+  promptTextCreator: (label: any) => label,
+  commaChoosesOption: false,
+};
+
+const sourceControl = {
+  name: 'source',
+  config: {
+    ...controlConfig,
+    clearable: false,
+    label: t('Source'),
+    description: t('Source for nodes of graph'),
+  },
+};
+
+const targetControl = {
+  name: 'target',
+  config: {
+    ...controlConfig,
+    clearable: false,
+    label: t('Target'),
+    description: t('Target for nodes of graph'),
+  },
+};
+
+const categoryControl = {
```

Comment:
This is used for coloring, right? How about just name this `Color by`?

Reply:
yeah,I think categoryControl seems more appropriate.

Reply:
Up to you. You can keep the control name `categoryControl`, so it is more consistent with ECharts options, but since it is now only used for color coding, to the end users maybe the control label can be `Color by` (or we should at least mention it in the tooltip).